pkgs/applications/audio/spotify-player/default.nix

```diff
@@ -30,31 +48,37 @@ rustPlatform.buildRustPackage rec {
 
   buildInputs = [
     openssl
-    alsa-lib
     dbus
     fontconfig
-    libsixel
-  ];
+  ]
+    ++ lib.optionals withSixel [ libsixel ]
+    ++ lib.optionals (withAudioBackend == "alsa") [ alsa-lib ]
```

Comment:
Probably can be merged with the `rodio` option.

Reply:
I think its best to have the audio backends exactly as specified in [librespot documentation](https://github.com/librespot-org/librespot/wiki/Compiling#general-dependencies)